Feature: Add ability to sort via entry created time

jloh opened the pull request:
Do you follow the guidelines?

- [x] I have tested my changes
- [x] I read this document: https://miniflux.app/faq.html#pull-request

Hi all. This PR adds the ability to change the default sort order of entries in the Miniflux UI to either be the time from the RSS feed or the time that Miniflux found/crawled the entry.

My use case for this is that I have several feeds that either publish articles backdated or without a timezone so often come in a lot further than the list than I expect.

I understand if this PR wont be merged however if it will be accepted I'll finish it off and do the other translations.

I'm very new to Go so apologies if something is wrong! Testing locally it works for me however.

jloh commented:
I've pushed the rest of the language files required. I admit I just ran the sentences through Google Translate 😬

jloh commented:
Have fixed the integration tests, didn't realise the client needed updating as well sorry!

jlelse commented on locale/translations/de_DE.json:
```diff
@@ -293,10 +293,13 @@
     "form.prefs.label.entry_swipe": "Wischgeste für Einträge auf dem Handy aktivieren",
     "form.prefs.label.show_reading_time": "Geschätzte Lesezeit für Artikel anzeigen",
     "form.prefs.label.custom_css": "Benutzerdefiniertes CSS",
+    "form.prefs.label.entry_order": "Eintragsbestellung",
```
Eintragsreihenfolge

jloh replied:
Thanks, updated!

jlelse commented:
> I've pushed the rest of the language files required. I admit I just ran the sentences through Google Translate grimacing

[DeepL](https://www.deepl.com/) probably gives better results. I guess "bestellen" in NL is a *strange* translation as well. (I don't speak NL, but it's a bit similar to German.)

fguillot reviewed (commented):
This feature might bring some confusion to the end-user because there is an existing option called "Entry Sorting" just above. Maybe it's about using better wording.

![image](https://user-images.githubusercontent.com/323546/120232185-e22b9a00-c207-11eb-828e-878e3a095b68.png)

Perhaps, something like "Sorting Criteria" or "Sorting Column" might be better. I don't know, just throwing some ideas.

Another approach is to use only one dropdown and combine both features in the web ui.

jloh commented on locale/translations/fr_FR.json:
```diff
@@ -293,10 +293,13 @@
     "form.prefs.label.entry_swipe": "Activer le geste de balayage sur les entrées sur mobile",
     "form.prefs.label.show_reading_time": "Afficher le temps de lecture estimé des articles",
     "form.prefs.label.custom_css": "CSS personnalisé",
+    "form.prefs.label.entry_order": "Ordre d'entrée",
```
I've pushed the updates for now.

jloh commented:
Yeah, agreed the title needs to change. I worry adding it to the existing dropdown and combining them could be even more complicated since there would be so many options.

I'm happy to go either way but I think a title change is best.

jloh commented:
Hey @fguillot. Apologies for letting this PR get stale, I've finally come back to it and rebased it on the latest code. I updated the title to be _Entry Sorting Column_ as suggested along with all the translations so it now looks like this:
<img width="180" alt="image" src="https://user-images.githubusercontent.com/3689532/127757031-a3755bee-dd85-42b9-8414-0120b4651807.png">

Along with a small text change to the actual dropdowns:

<img width="192" alt="image" src="https://user-images.githubusercontent.com/3689532/127757034-3a34410d-bcfc-4c4c-a287-f1f922acee14.png">

Let me know what you think!

jloh commented:
I've rebased this onto the current main branch so it doesn't have anything blocking and is good for review whenever.

fguillot commented on database/migrations.go:
```diff
@@ -555,4 +555,13 @@ var migrations = []func(tx *sql.Tx) error{
 	},
+	func(tx *sql.Tx) (err error) {
+		sql := `
+			ALTER TABLE integrations ADD COLUMN telegram_bot_enabled bool default 'f';
```
Looks like there is a mistake here:

```
-> Current schema version: 52
-> Latest schema version: 53
* Migrating to version: 53
[FATAL] [Migration v53] pq: column "telegram_bot_enabled" of relation "integrations" already exists
exit status 1
```

jloh commented:
Rebase went bad 🤦 now _actually_ fixed.

jloh commented:
Thankyou @fguillot!
